server/lib/settings/index.js

```diff
@@ -41,14 +41,48 @@ const getAllSettingFromDb = async () => {
   return settingsList;
 };
 
+function getSettingValueType(settingValue) {
+  switch (typeof settingValue) {
+    case "string":
+      return "string"
+    case "bigint":
+    case "number":
+      return "number"
+    case "object":
+      return "json"
+    case "boolean":
+      return "boolean"
+    default:
+      console.warn(`unsupported SettingValueType('${typeof settingValue}'), fallbacks to 'string'`)
+      return "string"
+  }
+}
+
 module.exports = {
   getSettingByKeyFromDb,
   getSettingsList: async () => {
     const settings = await getAllSettingFromDb();
     const settingsList = {...settingsDefaults};
-    settings.map((setting) => {
-      settingsList[setting.key] = setting.value;
+
+    let settingsToInsert = []
+
+    Object.keys(settingsList).map((defaultSettingKey) => {
+      if (settings.map((s) => s.key).includes(defaultSettingKey)) {
+        settingsList[defaultSettingKey] = settings.filter(s => s.key === defaultSettingKey)[0].value;
+      } else {
+        const defaultSettingToStore = Setting.build({
+            key: defaultSettingKey,
+            value: settingsList[defaultSettingKey].toString(),
+            type: getSettingValueType(settingsList[defaultSettingKey])
+        });
+        settingsToInsert.push(defaultSettingToStore.save());
```

Comment:
you can create array of objects and use blukCreate instead of promise.all

Reply:
done